tests/regression/CMakeLists.txt

```diff
@@ -1,9 +1,12 @@
-enable_language(Fortran) #TODO conditionally
-enable_language(C) #TODO conditionally
+include(${PROJECT_SOURCE_DIR}/cmake/detect_features.cmake)
+detect_fortran_compiler()
+detect_c_compiler()
 
-add_subdirectory(array)
-if(CPP_BINDGEN_GT_LEGACY)
-    add_subdirectory(array_gt_legacy)
-endif()
+if(CMAKE_Fortran_COMPILER_LOADED)
+    add_subdirectory(array)
+    if(CPP_BINDGEN_GT_LEGACY)
+        add_subdirectory(array_gt_legacy)
+    endif()
 
-add_subdirectory(simple)
+    add_subdirectory(simple)
```

Comment:
Is this correct? Shouldn't this be outside the if? (Ther are C-tests)

Reply:
Yes, thanks. Maybe I should move it inside the subfolders for everything, but then it is less visible. So for now I'll fix it but keep the logic for the other folders here.

Reply:
Looks good. I'll make a note to file a bug report to CMake when I have some free minutes (the issues with check_language)